metadata-ingestion/src/datahub/cli/docker_check.py

```diff
@@ -189,28 +157,32 @@ def to_exception(
 
 def check_docker_quickstart() -> QuickstartStatus:
     container_statuses: List[DockerContainerStatus] = []
-
     with get_docker_client() as client:
         containers = client.containers.list(
             all=True,
             filters=DATAHUB_COMPOSE_PROJECT_FILTER,
         )
+        if len(containers) == 0:
+            return QuickstartStatus([])
+
+        # load the expected containers from the docker-compose file
+        config_file = containers[0].labels.get(
+            "com.docker.compose.project.config_files"
+        )
+        all_containers = []
+        with open(config_file, "r") as config_file:
+            all_containers = yaml.safe_load(config_file).get("services", {}).keys()
 
+        existing_containers = set()
         # Check that the containers are running and healthy.
         container: docker.models.containers.Container
         for container in containers:
-            name = container.name
+            name = container.labels.get("com.docker.compose.service", container.name)
+            existing_containers.add(name)
             status = ContainerStatus.OK
-
-            if container.name not in (
-                REQUIRED_CONTAINERS + CONTAINERS_TO_CHECK_IF_PRESENT
-            ):
-                # Ignores things like "datahub-frontend" which are no longer used.
-                # This way, we only check required containers like "datahub-frontend-react"
-                # even if there are some old containers lying around.
+            if name not in all_containers:
                 continue
```

Comment:
just for clarity, keeping the comment around to explain how this filters out old containers would be useful